apiserver/pkg/http/client.go

```diff
@@ -84,7 +84,7 @@ func (krc *KuberayAPIServerClient) CreateComputeTemplate(request *api.CreateComp
 	}
 	computeTemplate := &api.ComputeTemplate{}
 	if err := krc.unmarshaler.Unmarshal(bodyBytes, computeTemplate); err != nil {
-		return nil, status, nil
+		return nil, status, fmt.Errorf("failed to unmarshal: %+w", err)
```

Comment:
This change is not a no-op (the error changes from nil to non-nil). Could you please either (1) check whether this change is acceptable, or (2) remove the change from this PR and open a follow-up PR after completing the investigation?

Reply:
Yes, this is a bug we should fix. I could open a separate PR

Reply:
I have removed the return value change, plan to make it in a separate to make no-op change vs bug fix cleaner.